syntax = "proto3";

package ecommerce;

service ProductService {
    rpc CreateProduct (Product) returns (ProductResponse);
    rpc GetProduct (ProductId) returns (ProductResponse);
    rpc UpdateProduct (Product) returns (ProductResponse);
    rpc DeleteProduct (ProductId) returns (ProductResponse);
    rpc ListProducts (Empty) returns (ProductList);
}

message Product {
    string id = 1;
    string name = 2;
    string description = 3;
    double price = 4;
}

message ProductId {
    string id = 1;
}

message ProductResponse {
    bool success = 1;
    string message = 2;
    Product product = 3;
}

message ProductList {
    repeated Product products = 1;
}

message Empty {}